wireup/ioc/util.py

```diff
@@ -85,28 +94,76 @@ def get_globals(obj: type[Any] | Callable[..., Any]) -> dict[str, Any]:
 T = TypeVar("T")
 
 
+def _eval_type_native(
+    value: typing.ForwardRef,
+    globalns: dict[str, Any] | None = None,
+    localns: dict[str, Any] | None = None,
+) -> Any:
+    """Evaluate a ForwardRef using the native typing._eval_type function.
+
+    This function handles version-specific differences in the _eval_type signature.
+    """
+
+    if _eval_type is None:
+        msg = "typing._eval_type is not available in this Python version."
+        raise RuntimeError(msg)
+
+    if sys.version_info >= (3, 12):
+        return _eval_type(value, globalns, localns, None)
```

Comment:
This version of eval_type has None has None as the default value of it's fourth parameter anyway so we can just not pass it. Also if we do that then we can merge these two branches and simplify the function.